pkg/controllermanager/controller/shoot/shoot_maintenance_control.go

```diff
@@ -145,88 +145,69 @@ func (c *defaultMaintenanceControl) Maintain(shootObj *gardenv1beta1.Shoot, key
+	updatedMachineImages, err := MaintainMachineImages(operation.Shoot.Info, operation.Shoot.CloudProfile, operation.Shoot.GetMachineImages())
 	if err != nil {
-		// continue execution to allow the kubernetes version update
+		// continue execution to allow the machine image version update
 		handleError(fmt.Sprintf("Could not maintain machine image version: %s", err.Error()))
 	}
 
-	var updateDefaultMachineImage func(s *gardenv1beta1.Cloud)
-	if defaultMachineImage != nil {
-		updateDefaultMachineImage = helper.UpdateDefaultMachineImage(operation.Shoot.CloudProvider, defaultMachineImage)
-	}
-
-	var updateWorkerMachineImages func(s *gardenv1beta1.Cloud)
-	if len(machineImages) > 0 {
-		updateWorkerMachineImages = helper.UpdateMachineImages(operation.Shoot.CloudProvider, machineImages)
-	}
-
-	updatedVersion, err := MaintainKubernetesVersion(operation.Shoot.Info, operation.Shoot.CloudProfile)
+	updatedKubernetesVersion, err := MaintainKubernetesVersion(operation.Shoot.Info, operation.Shoot.CloudProfile)
 	if err != nil {
 		// continue execution to allow the kubernetes version update
 		handleError(fmt.Sprintf("Could not maintain kubernetes version: %s", err.Error()))
 	}
 
-	// Check if the CloudProfile contains a newer Kubernetes patch version.
-	var updateKubernetesVersion func(s *gardenv1beta1.Kubernetes)
-	if updatedVersion != nil {
-		updateKubernetesVersion = func(s *gardenv1beta1.Kubernetes) { s.Version = *updatedVersion }
-	}
-
 	// Update the Shoot resource object.
-	_, err = kutil.TryUpdateShoot(c.k8sGardenClient.Garden(), retry.DefaultBackoff, shoot.ObjectMeta, func(s *gardenv1beta1.Shoot) (*gardenv1beta1.Shoot, error) {
+	_, err = kutil.TryUpdateShoot(c.k8sGardenClient.GardenCore(), retry.DefaultBackoff, shoot.ObjectMeta, func(s *gardencorev1alpha1.Shoot) (*gardencorev1alpha1.Shoot, error) {
 		if !apiequality.Semantic.DeepEqual(shootObj.Spec.Maintenance.AutoUpdate, s.Spec.Maintenance.AutoUpdate) {
 			return nil, fmt.Errorf("auto update section of Shoot %s/%s changed mid-air", s.Namespace, s.Name)
 		}
 
 		delete(s.Annotations, common.ShootOperation)
-
-		controllerutils.AddTasks(s.Annotations, common.ShootTaskDeployInfrastructure, common.ShootTaskDeployKube2IAMResource)
+		controllerutils.AddTasks(s.Annotations, common.ShootTaskDeployInfrastructure)
 		s.Annotations[common.ShootOperation] = common.ShootOperationReconcile
 
-		if updateDefaultMachineImage != nil {
-			updateDefaultMachineImage(&s.Spec.Cloud)
+		if updatedMachineImages != nil {
+			gardencorev1alpha1helper.UpdateMachineImages(shoot.Spec.Provider.Workers, updatedMachineImages)
 		}
-		if updateWorkerMachineImages != nil {
-			updateWorkerMachineImages(&s.Spec.Cloud)
-		}
-		if updateKubernetesVersion != nil {
-			updateKubernetesVersion(&s.Spec.Kubernetes)
+		if updatedKubernetesVersion != nil {
+			shoot.Spec.Kubernetes.Version = *updatedKubernetesVersion
```

Comment:
```suggestion
			s.Spec.Kubernetes.Version = *updatedKubernetesVersion
```

Please fix. Otherwise no k8 and machine image update are being conducted.